### ISSUE TITLE
How can I make Syntastic load a different checker based on existance of files in root?

### ISSUE THREAD

**Issue by yoshuawuyts** [scrooloose/syntastic#1327]
_Originally posted on [StackOverflow](http://stackoverflow.com/questions/28573553/how-can-i-make-syntastic-load-a-different-checker-based-on-existance-of-files-in) but also opening here for reference._

At work we use a different syntax checker than I do when working on open source. Is there a way to have [`Syntastic`](https://github.com/scrooloose/syntastic) specify a default checker, and change checkers if an `rc` file is found at the project root? 

Example: if `.eslintrc` is found use [`eslint`](https://github.com/eslint/eslint). If no `.eslintrc` is found, use [`standard`](https://github.com/feross/standard). 

Thanks!

**Comment by lcd047** [comment]
Yes, you can do something like this:
```vim
autocmd FileType javascript let b:syntastic_checkers = findfile('.eslintrc, '.;') != '' ? ['eslint'] : ['standard']
```

**Comment by yoshuawuyts** [comment]
@lcd047 thanks a lot!

**Comment by willian** [comment]
Does it still work?
I tried and I can't see any syntax error even if I force one. This is my `:SyntasticInfo` output:
```
Syntastic version: 3.8.0-3 (Vim 800, Darwin)
Info for filetype: javascript.jsx
Global mode: active
Filetype javascript.jsx is active
The current file will be checked automatically
Available checkers: javascript/eslint javascript/standard
Currently enabled checker: javascript/eslint
```

This is the `.eslintrc` I have in project's root directory:
```
{"extends": ["standard"], "parser": "babel-eslint"}
```

Another team member uses the WebStorm IDE and it requires a `.eslintrc` in order to have `standard` working, as described here: http://standardjs.com/webstorm.html

So, basically:

- My Syntastic configuration is using `eslint`
- The project's eslint configuration is using `stantard`

But I can't see any warning/error when I save or open a file with a `;`, for example:
![img1](https://cl.ly/3X2E0y2o0t1H/Image%202016-12-13%20at%2012.14.46%20PM.png)

If I change my configuration to use only `stantard` as my default checker I can see it working:
![img2](https://cl.ly/1b1H1H3T3R0J/Image%202016-12-13%20at%2012.16.28%20PM.png)

**Comment by lcd047** [comment]
Please open your test file, set `g:syntastic_debug` to 3, run the checker, run `:mes`, and post the output.  Please use copy & paste rather than screenshots.

**Comment by willian** [comment]
thanks @lcd047, debugging showed me that some packages was missing and that I should installed those globally :)

**Comment by willian** [comment]
Ok, I tested in a project that doesn't have any `.eslintrc` file and uses standard and did what @lcd047 suggested, this is the output:
```
Messages maintainer: Bram Moolenaar <Bram@vim.org>
"gulp/assets/javascripts/address_book/actions/address.js" 96L, 3587C
syntastic: 10.082935: g:syntastic_version = '3.8.0-3 (Vim 800, Darwin)'
syntastic: 10.083310: &shell = '/usr/local/bin/zsh', &shellcmdflag = '-c', &shellpipe = '2>&1| tee', &shellquote = '', &shellredir = '>%s 2>&1', &shelltemp = 1, &shellxquote = '', &autochdir
= 0, &shellxescape = ''
syntastic: 10.084188: UpdateErrors: default checkers
syntastic: 10.084465: CacheErrors: default checkers
syntastic: 10.085112: g:syntastic_aggregate_errors = 0
syntastic: 10.085294: getcwd() = '/Users/willian/Projects/PrintBear/denver/sticker-mule'
syntastic: 10.085880: CacheErrors: Invoking checker: javascript/eslint
syntastic: 10.086573: SyntasticMake: called with options: {'postprocess': ['guards'], 'errorformat': '%E%f: line %l\, col %c\, Error - %m,%W%f: line %l\, col %c\, Warning - %m', 'makeprg': 'e
slint -f compact gulp/assets/javascripts/address_book/actions/address.js'}
syntastic: 10.615169: system: command run in 0.528228s
syntastic: 10.615341: checker output: ['']
syntastic: 10.615613: raw loclist: [{'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': ''}]
syntastic: 10.615843: postprocess: [{'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': ''}]
syntastic: 10.616010: getLocList: checker javascript/eslint returned 0
syntastic: 10.616136: javascript/eslint raw: [{'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 0, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': ''}]
syntastic: 10.616321: quiet_messages filter: {}
syntastic: 10.616445: getLocList: checker javascript/eslint run in 0.530441s
syntastic: 10.616740: aggregated: {'_sorted': 0, '_name': '', '_owner': 1, '_columns': 1, '_rawLoclist': []}
```

**Comment by kevinoid** [comment]
I implemented checker detection rules for most JavaScript and TypeScript checkers for my own use (e.g. looking for `.eslintrc` with various extensions, `"eslintConfig"` in `package.json`, etc.).  I packaged it as a plugin ([kevinoid/syntastic-detect-checkers](https://github.com/kevinoid/syntastic-detect-checkers)) in case anyone else is interested in sharing checker detection rules.  If there is any interest in including the rules in Syntastic, I would be happy to help merge it.
